### PR TITLE
OSDOCS#5975: Custom RHCOS image for control plane and compute machines

### PR DESCRIPTION
**Pull request by mjpytlak**
Version(s):
4.14+

Issue:
This PR addresses [osdocs-5975](https://issues.redhat.com/browse/OSDOCS-5975).

Link to docs preview:

- [Additional Google Cloud Platform (GCP) configuration parameters](https://60348--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-customizations.html#installation-configuration-parameters-additional-gcp_installing-gcp-customizations)
- [Sample customized install-config.yaml file for GCP](https://60348--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-customizations.html#installation-gcp-config-yaml_installing-gcp-customizations)
- [Using a GCP Marketplace image](https://60348--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-customizations.html#installation-gcp-marketplace_installing-gcp-customizations)

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

**Comment by mjpytlak** [comment]
@AnnaZivkovic @r4f4 Draft docs ready for SME review.

@nirfarkas Do you ever get any feedback on your question about the use of the `Licenses` field? See your comment from 5/5/20023 in https://issues.redhat.com/browse/CORS-2445. Not sure if docs has to state anything explicitly.

**Comment by r4f4** [comment]
Replied on Slack, but for the record: the Licenses field will be incompatible with specifying boot images.

**Comment by mjpytlak** [comment]
> Replied on Slack, but for the record: the Licenses field will be incompatible with specifying boot images.

@r4f4 Thanks for the update. I cannot seem to find the reply in Slack. Based on this, I do not think there is anything else left to say, correct. If not, PTAL at the draft content when you have a moment.

@AnnaZivkovic PTAL as well. Thank you.

**Comment by AnnaZivkovic** [comment]
LGTM

**Comment by mjpytlak** [comment]
@jianli-wei Ready for QE review. PTAL.

File: modules/installation-configuration-parameters.adoc [hunk 1]
```diff
@@ -1331,6 +1331,22 @@ Additional GCP configuration parameters are described in the following table:
 |====
 |Parameter|Description|Values
 
+|`controlPlane.platform.gcp.osimage.project`
```
Comment (jianli-wei) [inline modules/installation-configuration-parameters.adoc:1334]: Shall `osimage.project` be `osImage.Project` instead? FYI the [PR](https://github.com/openshift/installer/pull/7215/).

File: modules/installation-configuration-parameters.adoc [hunk 2]
```diff
@@ -1331,6 +1331,22 @@ Additional GCP configuration parameters are described in the following table:
...
+|Optional. By default, the installation program downloads and installs the {op-system-first} image that is used to boot control plane machines. You can override the default behavior by specifying the location of a custom {op-system} image that the installation program is to use for control plane machines only.
+|String. The name of GCP project where the image is located.
+
+|`controlPlane.platform.gcp.osimage.name`
```
Comment (jianli-wei) [inline modules/installation-configuration-parameters.adoc:1338]: Similarly, shall `osimage.name` be `osImage.Name`?

**Review by jianli-wei — CHANGES_REQUESTED** [review]

File: modules/installation-gcp-config-yaml.adoc [hunk 3]
```diff
@@ -58,6 +58,9 @@ controlPlane: <2> <3>
       tags: <6>
       - control-plane-tag1
       - control-plane-tag2
+      osimage: <7>
+        project: example-project-name
+        name: example-image-name
```
Comment (jianli-wei) [inline modules/installation-gcp-config-yaml.adoc:63]: "osimage" => "osImage"?
"project" => "Project"?
"name" => "Name"?

File: modules/installation-gcp-config-yaml.adoc [hunk 4]
```diff
@@ -80,6 +83,9 @@ compute: <2> <3>
         tags: <6>
         - compute-tag1
         - compute-tag2
+        osimage: <7>
+          project: example-project-name
+          name: example-image-name
```
Comment (jianli-wei) [inline modules/installation-gcp-config-yaml.adoc:88]: Similar to the above, thanks!

File: modules/installation-gcp-config-yaml.adoc [hunk 5]
```diff
@@ -105,59 +111,62 @@ platform:
       tags: <6>
       - global-tag1
       - global-tag2
+      osimage: <7>
+        project: example-project-name
+        name: example-image-name
```
Comment (jianli-wei) [inline modules/installation-gcp-config-yaml.adoc:116]: Similar to the above, thanks!

File: modules/installation-gcp-config-yaml.adoc [hunk 6]
```diff
@@ -191,55 +200,56 @@ If you disable simultaneous multithreading, ensure that your capacity planning a
...
 <6> Optional: A set of network tags to apply to the control plane or compute machine sets. The `platform.gcp.defaultMachinePlatform.tags` parameter will apply to both control plane and compute machines. If the `compute.platform.gcp.tags` or `controlPlane.platform.gcp.tags` parameters are set, they override the `platform.gcp.defaultMachinePlatform.tags` parameter.
-<7> The cluster network plugin to install. The supported values are `OVNKubernetes` and `OpenShiftSDN`. The default value is `OVNKubernetes`.
+<7> Optional: A custom {op-system-first} that should be used to boot control plane and compute machines. The `project` and `name` parameters under `platform.gcp.defaultMachinePlatform.osimage` apply to both control plane and compute machines. If the `project` and `name` parameters under `controlPlane.platform.gcp.osimage` or `compute.platform.gcp.osimage` are set, they override the `platform.gcp.defaultMachinePlatform.osimage` parameters.
```
Comment (jianli-wei) [inline modules/installation-gcp-config-yaml.adoc:203]: Similar to the above, thanks!

File: modules/installation-gcp-marketplace.adoc [hunk 7]
```diff
@@ -4,45 +4,39 @@
...
 
-** `<installation_dir>/openshift/99_openshift-cluster-api_worker-machineset-0.yaml`
-** `<installation_dir>/openshift/99_openshift-cluster-api_worker-machineset-1.yaml`
-** `<installation_dir>/openshift/99_openshift-cluster-api_worker-machineset-2.yaml`
+.Procedure
 
-. In each file, edit the `.spec.template.spec.providerSpec.value.disks[0].image` property to reference the offer to use:
+. Edit the `compute.platform.gcp.osimage` parameters to specify the location of the GCP Marketplace image:
+** Set the `project` parameter to `redhat-marketplace-public`
+** Set the `name` parameter to one of the following offers:
```
Comment (jianli-wei) [inline modules/installation-gcp-marketplace.adoc:21]: Similar to the above, thanks!

**Review by jianli-wei — CHANGES_REQUESTED** [review]

**Comment by jianli-wei** [comment]
> Update the parameter names. Please take one more look.

@mjpytlak Sorry for the late reply, some minor suggestions, could you please update again? Thanks in advance!

**Comment by mjpytlak** [comment]
> Sorry for the late reply, some minor suggestions, could you please update again? Thanks in advance!

Thanks @jianli-wei I believe this should do it. PTAL.

**Comment by r4f4** [comment]
/hold
Patrick suggested in the PR the fields should be lowercase

**Comment by mjpytlak** [comment]
@r4f4 Thanks for info. Reverted all instances of `Project` and `Name` to `project` and `name`. PTAL.

**Comment by r4f4** [comment]
/lgtm
but let's wait a bit to make sure there will be no more changes in the Installer PR.

**Comment by mjpytlak** [comment]
@r4f4 PTAL and remove the merge hold, please. :smile:

**Comment by r4f4** [comment]
/hold cancel
My bad, forgot about this.

File: modules/installation-gcp-marketplace.adoc [hunk 8]
```diff
@@ -4,45 +4,39 @@
...
 :_content-type: PROCEDURE
 [id="installation-gcp-marketplace_{context}"]
-= Using a GCP Marketplace image
-If you want to deploy an {product-title} cluster using a GCP Marketplace image, you must create the manifests and edit the compute machine set definitions to specify the GCP Marketplace image.
+= Using the GCP Marketplace offering
 
-.Prerequisites
-
-* You have the {product-title} installation program and the pull secret for your cluster.
+Using the GCP Marketplace offering lets you deploy an {product-title} cluster, which is billed on pay-per-use basis (hourly, per core) through GCP, while still being supported directly by Red Hat.
```
Comment (nalhadef) [inline modules/installation-gcp-marketplace.adoc:9]: Need a nonbreaking space between "Red" and "Hat?"
image[
]
(https://redhat-documentation.github.io/supplementary-style-guide/#introduction)

**Comment by nalhadef** [comment]
One small comment, @mjpytlak.

**Comment by mjpytlak** [comment]
Thanks for catching the new requirement @nalhadef. Updated.

**Pull request merged**